lib/parity/environment.rb

```diff
@@ -37,8 +41,14 @@ def backup
     end
 
     def deploy
-      if deploy_to_heroku && run_migration?
+      pending_migrations = run_migration?
+
+      deployed = deploy_to_heroku
```

Comment:
The required ordering and caching of values in this method makes it surprisingly complex. I don't have any particularly great ideas for cleaning it up, but what about:

```ruby
skip_migrate = !run_migration?

if deploy_to_heroku
  skip_migrate || migrate
end
```